lib/ecto/adapters/mysql/connection.ex

```diff
@@ -0,0 +1,519 @@
+      Mariaex.Connection.start_link(opts)
+    end
+
+    def disconnect(conn) do
+      try do
+        Mariaex.Connection.stop(conn)
+      catch
+        :exit, {:noproc, _} -> :ok
+      end
+      :ok
+    end
+
+    def query(conn, sql, params, opts \\ []) do
+      case Mariaex.Connection.query(conn, sql, params, opts) do
+        {:ok, %Mariaex.Result{} = result} -> {:ok, Map.from_struct(result)}
+        {:error, %Mariaex.Error{}} =  err -> err
+      end
+    end
+
+    ## Transaction
+
+    def begin_transaction do
+      "BEGIN"
+    end
+
+    def rollback do
+      "ROLLBACK"
+    end
+
+    def commit do
+      "COMMIT"
+    end
+
+    def savepoint(savepoint) do
+      "SAVEPOINT " <> savepoint
+    end
+
+    def rollback_to_savepoint(savepoint) do
+      "ROLLBACK TO SAVEPOINT " <> savepoint
+    end
+
+    ## Query
+
+    alias Ecto.Query.SelectExpr
+    alias Ecto.Query.QueryExpr
+    alias Ecto.Query.JoinExpr
+
+    def all(query) do
+      sources = create_names(query)
+
+      from     = from(sources)
+      select   = select(query.select, query.distincts, sources)
+      join     = join(query.joins, sources)
+      where    = where(query.wheres, sources)
+      group_by = group_by(query.group_bys, sources)
+      having   = having(query.havings, sources)
+      order_by = order_by(query.order_bys, sources)
+      limit    = limit(query.limit, sources)
+      offset   = offset(query.offset, sources)
+      lock     = lock(query.lock)
+
+      assemble([select, from, join, where, group_by, having, order_by, limit, offset, lock])
+    end
+
+    def update_all(query, values) do
+      sources = create_names(query)
+      {table, name, _model} = elem(sources, 0)
+
+      zipped_sql = Enum.map_join(values, ", ", fn {field, expr} ->
+        "#{quote_name(field)} = #{expr(expr, sources)}"
+      end)
+
+      where = where(query.wheres, sources)
+      where = if where, do: " " <> where, else: ""
+
+      "UPDATE #{quote_name(table)} AS #{name} " <>
+      "SET " <> zipped_sql <>
+      where
+    end
+
+    def delete_all(query) do
+      sources = create_names(query)
+      {table, name, _model} = elem(sources, 0)
+
+      where = where(query.wheres, sources)
+
+      if where do
+        "DELETE FROM #{name} USING #{quote_name(table)} AS #{name} " <> where
+      else
+        "DELETE FROM #{quote_name(table)}"
+      end
+    end
+
+    # Returning isn't supported on MySQL
+    def insert(table, fields, _returning) do
+      values =
+        if fields == [] do
+          "() VALUES ()"
+        else
+          "(" <> Enum.map_join(fields, ", ", &quote_name/1) <> ") " <>
+          "VALUES (" <> Enum.map_join(1..length(fields), ", ", fn (_) -> "?" end) <> ")"
+        end
+
+      "INSERT INTO #{quote_name(table)} " <> values
+    end
+
+    def last_inserted(table, pk) do
+      "SELECT #{quote_name(pk)} FROM #{quote_name(table)} WHERE #{quote_name(pk)} = LAST_INSERT_ID()"
```

Comment:
Sure! Just for documentation purpose, this is the documentation for `LAST_INSERT_ID()`: http://dev.mysql.com/doc/refman/5.0/en/information-functions.html#function_last-insert-id